Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -0,0 +1,357 @@
+            if response.status_code not in {200, 201}:
+                return_error(f'Error in Microsoft authorization. Status: {response.status_code},'
+                             f' body: {self.error_parser(response)}')
+            body = response.json()
+        except Exception as e:
+            return_error(f'Error in Microsoft authorization: {str(e)}')
+
+        access_token = body.get('access_token', '')
+        expires_in = int(body.get('expires_in', 3595))
+
+        return access_token, expires_in
+
+    @staticmethod
+    def error_parser(error: requests.Response) -> str:
+        """
+
+        Args:
+            error (requests.Response): response with error
+
+        Returns:
+            str: string of error
+
+        """
+        try:
+            response = error.json()
+            inner_error = response.get('error', {})
+            if isinstance(inner_error, dict):
+                err_str = f"{inner_error.get('code')}: {inner_error.get('message')}"
+            else:
+                err_str = inner_error
+            if err_str:
+                return err_str
+            # If no error message
+            raise ValueError
+        except ValueError:
+            return error.text
+
+    @staticmethod
+    def epoch_seconds(d: datetime = None) -> int:
+        """
+        Return the number of seconds for given date. If no date, return current.
+
+        Args:
+            d (datetime): timestamp
+        Returns:
+             int: timestamp in epoch
+        """
+        if not d:
+            d = datetime.utcnow()
+        return int((d - datetime.utcfromtimestamp(0)).total_seconds())
+
+    @staticmethod
+    def get_encrypted(content: str, key: str) -> str:
+        """
+        Encrypts content with encryption key.
+        Args:
+            content: Content to encrypt
+            key: encryption key from oproxy
+
+        Returns:
+            timestamp: Encrypted content
+        """
+
+        def create_nonce():
+            return os.urandom(12)
+
+        def encrypt(string, enc_key):
+            """
+            Encrypts string input with encryption key.
+            Args:
+                string: String to encrypt
+                enc_key: Encryption key
+
+            Returns:
+                bytes: Encrypted value
+            """
+            # String to bytes
+            enc_key = base64.b64decode(enc_key)
+            # Create key
+            aes_gcm = AESGCM(enc_key)
+            # Create nonce
+            nonce = create_nonce()
+            # Create ciphered data
+            data = string.encode()
+            ct = aes_gcm.encrypt(nonce, data, None)
+            return base64.b64encode(nonce + ct)
+
+        now = MicrosoftClient.epoch_seconds()
+        encrypted = encrypt(f'{now}:{content}', key).decode('utf-8')
+        return encrypted
+
+    @staticmethod
+    def _add_info_headers() -> Dict[str, str]:
+        # pylint: disable=no-member
```

Comment:
why?

Reply:
`callingContext` is not a member of `demisto`.